### DISCUSSION TITLE
raytranslate losing point cloud precision

### DISCUSSION THREAD

**Discussion by ShukhratSh** [csiro-robotics/raycloudtools#10] [General]
Hi Thomas,
I apologize for reaching out with multiple questions. I have imported the point cloud data and applied segmentation algorithms to generate tree segments. My current objective is to compare the segmentation results with manually segmented trees to evaluate the performance of the segmentation algorithm. To achieve this, I need to translate the point cloud back to its original xyz coordinates using the raytranslate tool.

Here is the command I used.

```
rayimport *.las ray 0,0,0 --max_intensity 0 --remove_start_pos
readLas: filename: point_cloud.las
offset to remove:      530812 4.26064e+06     503.193
read and process 302 / 303 60.558s  
loaded point_cloud.las with 302,991,538 points
start position:     530,812 4.26064e+06     503.193 removed from all points

raytranslate *segmented.ply 530812,4260640,503.193
```
After translation y coordinates of the segmented rounded off so it lost the precision as shown in the screenshot. 
Any ideas why this might be happening?

![image](https://github.com/csiro-robotics/raycloudtools/assets/11915606/cd064e3e-0930-4045-9a84-0aadd6660db0)

Thank you

**Comment by thomas-lowe** [comment]
Hi Shukhrat,

 by default raycloudtools stores the x,y,z as floats in the .ply file. You can swap it to storing as doubles using the DOUBLE_RAYS build option. e.g. cmake .. -DDOUBLE_RAYS=ON

**Comment by ShukhratSh** [comment]
Hi Thomas, Thank you, I couldn't fix it yet. Do you know what a warning message means when importing the .las file in the following command? I suspect this is causing the round off the Y coordinates.
```
rayimport *_1.las ray 0,0,0 --max_intensity 0
readLas: filename: c6_tls_p5_1.las
WARNING: very large point location at: 0:     530,979 4.26045e+06      379.32, suspicious
read and process 410 / 412 76.412s  
loaded c6_tls_p5_1.las with 411,088,416 points
```

**Reply by thomas-lowe** [reply]
Hi ShukhratSh,
 that warning only happens when you are storing the ray end points with floats. So for some reason your cmake .. -DDOUBLE_RAYS=ON didn't work. 
 I use ccmake (you'll need to install it). You run ccmake .. and it shows you visually what all the build arguments are. Here's mine, confirming that DOUBLE_RAYS is on:
![ccmake](https://github.com/csiro-robotics/raycloudtools/assets/44390512/a65e9b19-63e5-49cb-8d31-4c94e806a91c)
